### DISCUSSION TITLE
Need to show error when a user types 13 for the month segment in useDateSegment

### DISCUSSION THREAD

**Discussion by TENsaga** [adobe/react-spectrum#5198] [Q&A]
When a user types in 13 for the month date segment it will show 1, then the 3 will erase the 1 (invalid month) and just display 3 for March. We've been asked to show an error when this happens letting the user know that 13 is an invalid month. I can't seem to find an easy way to do this and am wondering if I'm missing an obvious way to accomplish this? 

Here's the react-aria component/state management:

https://react-spectrum.adobe.com/react-aria/useDateField.html
https://react-spectrum.adobe.com/react-stately/useDateFieldState.html

Thanks for the help.

**Comment by reidbarber** [comment]
I don't know if there is a good way to do this. I think you could maybe listen for keydown events and handle this sort of thing independently from the component. That solution is fairly hackish, but I don't think we would want to add any sort of API for this sort of thing either.

**Reply by TENsaga** [reply]
Hey thanks for the reply, bit of a relief to know there's no easy way to do this after spending some time digging into it. I'll let my team know.
